src/twin/system_info.rs

```diff
@@ -118,4 +204,112 @@ impl SystemInfo {
+            return Ok(());
+        };
+
+        let mut disk_total = 0;
+        let mut disk_used = 0;
+        for disk in self.sysinfo_disk.list() {
+            if disk.name().to_str() == Some("/dev/omnect/data") {
+                disk_total = disk.total_space();
+                disk_used = disk.total_space() - disk.available_space();
+                break;
+            }
+        }
+
+        let mut metric_list = vec![
+            Metric::new(
+                time.clone(),
+                "cpu_usage",
+                self.sysinfo_memory.global_cpu_usage() as f64,
+                hostname.clone(),
+                None,
+            ),
+            Metric::new(
+                time.clone(),
+                "memory_used",
+                self.sysinfo_memory.used_memory() as f64,
+                hostname.clone(),
+                None,
+            ),
+            Metric::new(
+                time.clone(),
+                "memory_total",
+                self.sysinfo_memory.total_memory() as f64,
+                hostname.clone(),
+                None,
+            ),
+            Metric::new(
+                time.clone(),
+                "disk_used",
+                disk_used as f64,
+                hostname.clone(),
+                None,
+            ),
+            Metric::new(
+                time.clone(),
+                "disk_total",
+                disk_total as f64,
+                hostname.clone(),
+                None,
+            ),
+        ];
+
+        for component in self.sysinfo_components.iter() {
+            metric_list.push(Metric::new(
+                time.clone(),
+                "temp",
+                component.temperature() as f64,
+                hostname.clone(),
+                Some(format!("{}", component.label())),
+            ));
+        }
+
+        match serde_json::to_vec(&metric_list) {
+            Ok(json) => {
+                match IotMessage::builder()
+                    .set_body(json)
+                    .set_content_type("application/json")
+                    .set_content_encoding("utf-8")
+                    .set_output_queue("metrics")
+                    .build()
+                {
+                    Ok(msg) => {
+                        let Some(tx) = &self.tx_outgoing_message else {
+                            warn!("metrics_handler: skip since tx_outgoing_message is None");
+                            return Ok(());
+                        };
+
+                        let _ = tx.send(msg).await;
+                        Ok(())
+                    }
+                    Err(e) => {
+                        warn!("telemetry message could not be transmitted: {e}");
+                        return Ok(());
+                    }
+                }
+            }
+            Err(e) => {
+                warn!("metrics list could not be converted to vector: {e}");
+                return Ok(());
+            }
+        }
+    }
```

Comment:
I would prefer let `let Ok(json) = ... else ...` instead of match

further we should fail on all error cases and log an error.

we should also have an `info!` log for the success case somewhere

Reply:
this is imho more readable:
```
        let json = serde_json::to_vec(&metric_list)
            .context("metrics list could not be converted to vector:")?;

        let msg = IotMessage::builder()
            .set_body(json)
            .set_content_type("application/json")
            .set_content_encoding("utf-8")
            .set_output_queue("metrics")
            .build()
            .context("telemetry message could not be transmitted")?;

        tx.send(msg).await?;

        info!("metrics: telemetry message transmitted");

        Ok(())
```

Reply:
yes looks much cleaner and readable.

Reply:
done